backend/UserApp/views.py

```diff
@@ -124,62 +127,97 @@ def get(self, request):
         return TemplateResponse(request, self.template_name, context={})
 
 class UserUpdateUsernameView(APIView):
-    def put(self, request, username):
+    def put(self, request):
+        user_id = get_user_id(request)
+        new_username = request.data.get('new_username', None)
+        old_username = request.data.get('old_username', None)
+        
         try:
-            user = User.objects.get(username=username)
+            user = User.objects.get(id=user_id)
         except User.DoesNotExist:
-            return Response(
-                {'error': 'User not found'},
-                status=status.HTTP_404_NOT_FOUND
-            )
-
-        new_username = request.data.get('new_username', None)
+                return Response({
+                'success': False,
+                'message': 'User not found.'
```

Comment:
The success field isn't necessary anymore because the frontend can verify if the request succeeded by checking the status with response.ok.


Also just a small thing, but can you change the key that says `message` to be `error` instead and then don't send any message field on success? Just want some consistency for the frontend people.